### ISSUE TITLE
Switching Scala version with ++ modifies crossScalaVersions (1.0 regression)

### ISSUE THREAD

**Issue by ghik** [sbt/sbt#3495]
## steps

1. Create a project with more than one `crossScalaVersions`
1. Open SBT shell and inspect `crossScalaVersions` - it should list versions specified in `build.sbt`
1. Switch Scala version using `++`, e.g. `++2.11.11`
1. Inspect `crossScalaVersions` again - it should list the same versions as before, but it lists only the one consistent with current `scalaVersion`

```
scala-commons$ sbt
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /home/ghik/idea-workspace/scala-commons/project
[info] Loading settings from build.sbt,version.sbt ...
[info] Set current project to commons (in build file:/home/ghik/idea-workspace/scala-commons/)
[info] sbt server started at 127.0.0.1:4456
sbt:commons> commons-core/scalaVersion
[info] 2.12.3
sbt:commons> commons-core/crossScalaVersions
[info] * 2.11.11
[info] * 2.12.3
sbt:commons> ++2.11.11
[info] Setting Scala version to 2.11.11 on 13 projects.
[info] Reapplying settings...
[info] Set current project to commons (in build file:/home/ghik/idea-workspace/scala-commons/)
sbt:commons> commons-core/scalaVersion
[info] 2.11.11
sbt:commons> commons-core/crossScalaVersions
[info] * 2.11.11
sbt:commons>
```

## problem

`crossScalaVersions` is filtered when switching Scala version with `++`

## expectation

`crossScalaVersions` should not be modified when switching Scala version

## notes

This bug has a very nasty consequence - all cross-building commands starting with `+` (e.g. `+compile`) no longer work because after `crossScalaVersions` had been filtered it is impossible to switch again to any other Scala version using `++`.

sbt version: 1.0.1

**Comment by dwijnand** [comment]
@ghik I can't reproduce what you're seeing. Something's amiss. This is the build file I'm using:

```scala
crossScalaVersions in ThisBuild := List("2.10.6", "2.11.11", "2.12.3")
      scalaVersion in ThisBuild := "2.12.3"

val `commons-core` = project settings (
  crossScalaVersions := List("2.11.11", "2.12.3")
)
```

and this is the sbt shell session:

```
[info] Set current project to t-cross013 (in build file:/s/t-cross013/)
> show commons-core/scalaVersion
[info] 2.12.3
> show commons-core/crossScalaVersions
[info] * 2.12.3
> reload
[info] Loading settings from global-plugins.sbt,idea.sbt ...
[info] Loading global plugins from /Users/dnw/.dotfiles/.sbt/1.0/plugins
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /s/t-cross013/project
[info] Loading settings from build.sbt ...
[info] Set current project to t-cross013 (in build file:/s/t-cross013/)
> ++2.11.11
[info] Setting Scala version to 2.11.11 on 2 projects.
[info] Reapplying settings...
[info] Set current project to t-cross013 (in build file:/s/t-cross013/)
> show commons-core/scalaVersion
[info] 2.11.11
> commons-core/crossScalaVersions
[info] * 2.11.11
[info] * 2.12.3
```

**Comment by ghik** [comment]
@dwijnand it seems that the problem appears when `crossScalaVersions` is defined only `in ThisBuild`, e.g.

```
crossScalaVersions in ThisBuild := List("2.10.6", "2.11.11", "2.12.3")
scalaVersion in ThisBuild := "2.12.3"

val `commons-core` = project
```

then:
```
$ sbt
[info] Loading project definition from /home/ghik/idea-workspace/sbtbug/project
[info] Loading settings from build.sbt ...
[info] Set current project to sbtbug (in build file:/home/ghik/idea-workspace/sbtbug/)
[info] sbt server started at 127.0.0.1:4712
sbt:sbtbug> commons-core/crossScalaVersions
[info] * 2.10.6
[info] * 2.11.11
[info] * 2.12.3
sbt:sbtbug> commons-core/scalaVersion
[info] 2.12.3
sbt:sbtbug> ++2.11.11
[info] Setting Scala version to 2.11.11 on 2 projects.
[info] Reapplying settings...
[info] Set current project to sbtbug (in build file:/home/ghik/idea-workspace/sbtbug/)
sbt:sbtbug> commons-core/scalaVersion
[info] 2.11.11
sbt:sbtbug> commons-core/crossScalaVersions
[info] * 2.11.11
```

**Comment by dwijnand** [comment]
Thanks @ghik, I can reproduce that.

For the record this is a regression in behaviour from sbt 0.13, where it works as intended (notice the difference in output):

```
> ++2.11.11
[info] Setting version to 2.11.11
[info] Reapplying settings...
[info] Set current project to t-crossplusplus (in build file:/s/t-crossplusplus/)
> commons-core/crossScalaVersions
[info] * 2.10.6
[info] * 2.11.11
[info] * 2.12.3
```

**Comment by eed3si9n** [comment]
Fixed in #3526

**Comment by lukexingdbmgr** [comment]
I am new to kafka, meeting a similar issue. I probably missing something. Could anybody kindly give some help ?

1. git clone latest https://github.com/apache/kafka.git
2. cd kafka/
3. gradle 


luke@luke-VirtualBox:~/kafka/kafka$ gradle
Building project 'core' with Scala version 2.11.11

FAILURE: Build failed with an exception.

* Where:
Build file '/home/luke/kafka/kafka/build.gradle' line: 614

* What went wrong:
A problem occurred evaluating root project 'kafka'.
> /home/luke/kafka/kafka/docs/generated/protocol_errors.html (No such file or directory)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2.022 secs


luke@luke-VirtualBox:~/kafka/kafka$ gradle -version

------------------------------------------------------------
Gradle 3.0
------------------------------------------------------------

Build time:   2016-08-15 13:15:01 UTC
Revision:     ad76ba00f59ecb287bd3c037bd25fc3df13ca558

Groovy:       2.4.7
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_131 (Oracle Corporation 25.131-b11)
OS:           Linux 4.10.0-33-generic i386

luke@luke-VirtualBox:~/kafka/kafka$ java -version
openjdk version "1.8.0_131"
OpenJDK Runtime Environment (build 1.8.0_131-8u131-b11-2ubuntu1.17.04.3-b11)
OpenJDK Server VM (build 25.131-b11, mixed mode)
